File: node.js/백준/Bronze/1546. 평균/평균.js
'use strict';
const fs = require('fs');

let input = fs.readFileSync('/dev/stdin').toString().trim().split(/\s+/);
let n = Number(input[0]);
let scores = input.slice(1, n + 1).map(Number);


let maxScore = Math.max(...scores);


let sum = scores.reduce((acc, score) => acc + (score / maxScore * 100), 0);


let average = sum / n;
console.log(average);

File: C99/백준/Bronze/1546. 평균/평균.c
#include <stdio.h>

int main() {
    int n;
    double scores[1000]; // 점수 최대 1000개 저장 가능
    double max = 0.0, sum = 0.0;

    // 점수 개수 입력
    scanf("%d", &n);

    // 점수 입력 및 최댓값 찾기
    for (int i = 0; i < n; i++) {
        scanf("%lf", &scores[i]);
        if (scores[i] > max) {
            max = scores[i];
        }
    }

    // 점수 조작해서 합계 구하기
    for (int i = 0; i < n; i++) {
        scores[i] = (scores[i] / max) * 100;
        sum += scores[i];
    }

    // 새로운 평균 출력
    printf("%.2f\n", sum / n);

    return 0;
}